src/installer/image/installer.go

```diff
@@ -48,6 +48,11 @@ func (installer ImageInstaller) ImageDigest() string {
 func (installer *ImageInstaller) InstallAgent(targetDir string) (bool, error) {
 	log.Info("installing agent from image")
 
+	err := installer.fs.Mkdir(installer.props.PathResolver.AgentSharedBinaryDirBase(), common.MkDirFileMode)
```

Comment:
will this return an error in case the Dir already exists?

Reply:
`Mkdir` yes
`MkdirAll` no 😅